craft_parts/sources/local_source.py

```diff
@@ -36,14 +41,19 @@ def __init__(self, *args, copy_function=file_utils.link_or_copy, **kwargs):
         self.source_abspath = os.path.abspath(self.source)
         self.copy_function = copy_function
 
-        ignore_patterns = [
-            self._dirs.parts_dir.name,
-            self._dirs.stage_dir.name,
-            self._dirs.prime_dir.name,
-            "*.snap",  # FIXME: this should be specified by the application
-        ]
+        if self._dirs.work_dir.resolve() == Path.cwd():
```

Comment:
Would this break if cwd is not the project dir?  Perhaps this should instead check to see if self.source_abspath matches workdir?

Reply:
Good catch! Changed to check if workdir matches source dir, and fixed an inconsistent test which placed workdir outside source dir but expected it to be inside.